### DISCUSSION TITLE
Content hash not matching file hash

### DISCUSSION THREAD

**Discussion by LorenDorez** [webpack/webpack#17895] [Q&A]
Im using webpack v5 to add content hashes to the file name. However when i try to compare that hash to the File Hash generated from external tools they dont match. Does `[contentHash]` add data outside of the actual file contents?

*Webpack.config*
output: {
            filename: '[name].js',
            chunkFilename: 'chunks/[name].[contenthash].js?v=[contenthash]',
            hashFunction: 'sha256',
            hashDigest: 'base64url',
            hashDigestLength: 500,
        },

*Sample File Name*
`Components-Shared-Modals-FeedbackModal-jsx.jLTegxjMB4UskmwXWx6-T94R71wp7ftQyCc5GUlxVtk.js`

*Sample PS Script and results*
I know this one be 100% exact as Base64 and base64url varies slight for special characters (like +,\, etc..) but you can still see the hash are totally different.
```
# Path to File
$file = '\Components-Shared-Modals-FeedbackModal-jsx.jLTegxjMB4UskmwXWx6-T94R71wp7ftQyCc5GUlxVtk.js'

# Get the hash of the file
$hash = Get-FileHash $file -Algorithm SHA256

# Convert hash to bytes
$hashBytes = [System.Text.Encoding]::UTF8.GetBytes($hash.Hash)

# Convert byte to base64url
[System.Convert]::ToBase64String($hashBytes)
```

Output: `REQ5Q0E3QzJFNkVBNjJDQjg3QTk1QjdCOTJGMzU1ODExOTc4OUYzMERBNTZEQTRGMkZGRTdCRjRBOTBGNjU0Qw==`

**Comment by ErikLindroos** [comment]
You seem to be generating the hash as a hexstring and then converting that to bytes. Surely you mean to generate the hash as bytes from the onset?

**Comment by LorenDorez** [comment]
Right now im simply trying to understand and mimic the internal way Webpack is doing this. The goal is to have a better understanding of how Webpack generates this so i can create a custom Hasher to match our ASP.NET MVC Bundling Urls that use cache-busting by adding a querystring param based on a hash of the file it sees when creating the bundle.

But i need a better understand of how Webpack is doing the hashing before i can even try to create a custom one. So i figured if i can get a Powershell version to match Webpack then i would have a better understanding of how its doing that to create my custom hasher.

MVC Basically does the following
1. Read the file to string
2. Converts string to bytes using `Encoding.Unicode.GetBytes()`
3. Hashes that to `SHA256`
4. To Base64 with URL support

[a comment by an automated account is not shown]

**Comment by alexander-akait** [comment]
Can you provide reproducible example? Thank you

**Reply by LorenDorez** [reply]
Yeah I'll try and get you some example this weekend. Basically we use react for our front end and bundle using we pack but all our back end JS and CSS is served via ASP.Net MVC. It adds a content hash to the bundle urls for cache busting. I need them to essentially match so I don't have to dump the server side urls down to the client simply to override the get chunk url calls in a round-about way

**Reply by alexander-akait** [reply]
Feel free to ping me, I will help
